agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -1672,3 +1673,75 @@ func TestAppMeshCNIPlugin(t *testing.T) {
 
 	assert.Equal(t, 42, exitCode, fmt.Sprintf("Expected exit code of 42; got %d", exitCode))
 }
+
+// TestTrunkENIAttachDetachWorkflow tests that when ENI trunking is enabled, the Trunk ENI is attached
+// when container instance reaches ACTIVE status, and it's detached when the container instance is deregistered
+func TestTrunkENIAttachDetachWorkflow(t *testing.T) {
+	asyncWaitDuration := 30 * time.Second
+
+	RequireInstanceTypes(t, []string{"c5", "m5"})
+
+	// Enable ENI Trunking account setting
+	putAccountSettingInput := ecsapi.PutAccountSettingInput{
+		Name:  aws.String("awsvpcTrunking"),
+		Value: aws.String("enabled"),
+	}
+	_, err := ECS.PutAccountSetting(&putAccountSettingInput)
+	require.NoError(t, err)
+
+	existingMacs, err := GetNetworkInterfaceMacs()
+	require.NoError(t, err)
+	existingInterfaceCount := len(existingMacs)
+
+	agentOptions := &AgentOptions{
+		EnableTaskENI: true,
+	}
+	os.Setenv("ECS_FTEST_FORCE_NET_HOST", "true")
+
+	agent := RunAgent(t, agentOptions)
+	defer func() {
+		// Cleanup needs to happen regardless of what happened elsewhere
+		defer agent.TestCleanup()
+
+		err := agent.StopAgent()
+		require.NoError(t, err)
+
+		_, err = ECS.DeregisterContainerInstance(&ecsapi.DeregisterContainerInstanceInput{
+			Cluster:           &agent.Cluster,
+			ContainerInstance: &agent.ContainerInstanceArn,
+			Force:             aws.Bool(true),
+		})
+		require.NoError(t, err)
+
+		// Wait and verify that the Trunk ENI is detached after deregistration
+		err = WaitNetworkInterfaceCount(existingInterfaceCount, asyncWaitDuration)
+		assert.NoError(t, err)
+	}()
+	// TODO: when merging the changes, change to 1.27.1
+	agent.RequireVersion(">=1.26.0")
```

Comment:
I see this same `agent.RequireVersion()` up on line 803.  could this be declared once up top as a const?

Reply:
i think different functional tests may require different version of agents, so this probably shouldn't be declared globally. the agent version check enables us to run functional tests across different version of agents without changing the test code